src/rgw/rgw_op.cc

```diff
@@ -4824,33 +4733,25 @@ void RGWDeleteObj::execute()
     }
 
     if (!ver_restored) {
+      uint64_t epoch;
```

Comment:
in most cases, `get_system_versioning_params()` won't write to `epoch`, so this value remains uninitialized. this turned out to be the root cause of https://tracker.ceph.com/issues/47240. please see https://github.com/ceph/ceph/pull/36951 for the fix